### DISCUSSION TITLE
BelongsTo<typeof Category> avec Adonisjs 6

### DISCUSSION THREAD

**Discussion by LeFlow** [adonisjs/core#4449] [General]
Hi

I'm training on Adonisjs 6, but following an example based on Adonisjs 5.

In a model, the annotation is used
```
   @belongsTo(() => Category)
   public category: BelongsTo<typeof Category>
```

I imported:
`{ BaseModel, belongsTo, column } from '@adonisjs/lucid/orm'`
BelongsTon' is not recognized.

In this example, it is imported by
`'@adonisjs/lucid/orm'`

But it does not work.
My IDE offers to import it from
`'@adonisjs/lucid/types/relations'`

Despite everything, I have an error at the level of
`category: BelongsTo`

I don't know, it's quite explicit, but thanks anyway in advance (only if you've tried to understand).

TY

**Comment by holoyan** [comment]
Hi @LeFlow 


`@belongsTo(() => Category)` anotation is different from `BelongsTo` return type so you should also import the 'type' as well

`import {BelongsTo} from "@adonisjs/lucid/types/relations";`

**Comment by LeFlow** [comment]
II had tried `import {BelongsTo} from "@adonisjs/lucid/types/relations";`
But the problem is the same

**Comment by holoyan** [comment]
it will make clear if you put whole model, including import statments to better understood what's going on

**Comment by LeFlow** [comment]
```
import { DateTime } from 'luxon'
import { BaseModel, belongsTo, column } from '@adonisjs/lucid/orm'
import Category from './category.js'
import { BelongsTo } from '@adonisjs/lucid/types/relations'

export default class Post extends BaseModel {
  @column({ isPrimary: true })
  declare id: number

  @column()
  declare title: string

  @column()
  declare content: string

  @column()
  declare online: boolean

  @column()
  declare thumbnail: string | null

  @belongsTo(() => Category)
  public category: BelongsTo<typeof Category>

  @column.dateTime({ autoCreate: true })
  declare createdAt: DateTime | null

  @column.dateTime({ autoCreate: true, autoUpdate: true })
  declare updatedAt: DateTime | null
}
```
and 

```
import { DateTime } from 'luxon'
import { BaseModel, column } from '@adonisjs/lucid/orm'

export default class Category extends BaseModel {
  @column({ isPrimary: true })
  declare id: number

  @column()
  declare name: string

  @column.dateTime({ autoCreate: true })
  declare createdAt: DateTime

  @column.dateTime({ autoCreate: true, autoUpdate: true })
  declare updatedAt: DateTime
}
```

**Reply by tomgobich** [reply]
What does the error you're seeing say? Is it...
> A type referenced in a decorated signature must be imported with 'import type' or a namespace import when 'isolatedModules' and 'emitDecoratorMetadata' are enabled

If so, you just need to specify the import is a type-only import, note the `type` directly after import:
```ts
import type { BelongsTo } from '@adonisjs/lucid/types/relations'
```
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html

**Comment by LeFlow** [comment]
Very good.

TY All.
